src/main/java/net/pms/dlna/DLNAResource.java

```diff
@@ -3991,53 +3842,438 @@ protected void setDiscovered(boolean discovered) {
+		}
+
+		if (!hasSubtitles(forceRefresh)) {
+			return null;
+		}
+
+		/*
+		 * Check for external and internal subtitles matching the user's
+		 * language preferences
+		 */
+		DLNAMediaSubtitle matchedSub = null;
+		boolean useExternal = deviceSpecificConfiguration.isAutoloadExternalSubtitles();
+		String audioSubLanguages = deviceSpecificConfiguration.getAudioSubLanguages();
+		if (isBlank(audioLanguage) || isBlank(audioSubLanguages)) {
+			// Not enough information to do a full audio/subtitles combination search, only use the preferred subtitles
+			LOGGER.trace("Searching for subtitles without considering audio language");
+			ArrayList<DLNAMediaSubtitle> candidates = new ArrayList<>();
+			for (DLNAMediaSubtitle subtitles : media.getSubtitleTracksList()) {
+				if (subtitles.isExternal()) {
+					if (useExternal) {
+						candidates.add(subtitles);
+					}
+				} else {
+					candidates.add(subtitles);
+				}
+			}
+			if (!candidates.isEmpty()) {
+				matchedSub = SubtitleUtils.findPrioritizedSubtitles(candidates, renderer, false);
+				if (matchedSub != null) {
+					LOGGER.trace(
+						"Matched {} subtitles track: {}",
+						matchedSub.isExternal() ? "external" : "internal",
+						matchedSub
+					);
+					return matchedSub;
+				}
+			}
+		} else {
+			// Do a full audio/subtitles combination search
+			StringTokenizer st = new StringTokenizer(audioSubLanguages.toLowerCase(Locale.ROOT), ";");
+			while (st.hasMoreTokens()) {
+				String pair = st.nextToken();
+				int commaPos = pair.indexOf(',');
+				if (commaPos > -1) {
+					String audio = pair.substring(0, commaPos).trim();
+					String sub = pair.substring(commaPos + 1).trim();
+					LOGGER.trace(
+						"Searching for a match for audio language \"{}\" with audio \"{}\" and subtitle \"{}\"",
+						audioLanguage,
+						audio,
+						sub
+					);
+
+					if ("*".equals(audio) || DLNAMediaLang.UND.equals(audio) || Iso639.isCodesMatching(audio, audioLanguage)) {
+						boolean anyLanguage = "*".equals(sub) ||  DLNAMediaLang.UND.equals(sub);
+						if ("off".equals(sub)) {
+							if (deviceSpecificConfiguration.isForceExternalSubtitles()) {
+								// Ignore the "off" language for external subtitles if force external subtitles is enabled
+								ArrayList<DLNAMediaSubtitle> candidates = new ArrayList<>();
+								for (DLNAMediaSubtitle subtitles : media.getSubtitleTracksList()) {
+									if (subtitles.isExternal()) {
+										candidates.add(subtitles);
+									}
+								}
+								if (!candidates.isEmpty()) {
+									matchedSub = SubtitleUtils.findPrioritizedSubtitles(candidates, renderer, true);
+									LOGGER.trace(
+										"Ignoring the \"off\" language because external subtitles are enforced, returning: {}",
+										matchedSub
+									);
+									return matchedSub;
+								}
+							} else {
+								LOGGER.trace(
+									"Not looking for non-forced subtitles since they are \"off\" for audio language \"{}\"",
+									audio
+								);
```

Comment:
@Nadahar there should be added `break;` to break the `while` loop and skip parsing subtitles when they are "off". 
@SubJunk you were right that subs are used also in case when they are "off". I made the testing video and restested it.

Reply:
@valib Don't make a commit, I already have some unpushed changes. I'm not sure if I understand the "break" though, where exactly do you mean it should be?

Reply:
@Nadahar just after the line 4158. It breaks the while loop which causes to continue parsing for the next audio/subs combination like ```*,eng```